### PR TITLE
Feat/meeting card container/create

### PR DESCRIPTION
**Pull request by InteractiveTimmy**
# Description

The scope of the changes in this pull request are to create the core `<MeetingCardContainer />` component that will be utilized for all upcoming **Meeting Container** components. This component can be later used as the root for `<MeetingListItem />` container once the **Meeting Container** components are generated.

**Note** - The changes in this pull request also allow for localized CSS Variables to be created to help reduce code footprint and improve readability when multiple variations of a component utilize similar values from our design tokens.

## Screenshots

Screen Shot 2022-03-29 at 3 58 26 PM
Screen Shot 2022-03-29 at 3 57 35 PM
Screen Shot 2022-03-29 at 3 57 39 PM
Screen Shot 2022-03-29 at 3 57 48 PM
Screen Shot 2022-03-29 at 3 57 59 PM
Screen Shot 2022-03-29 at 3 58 05 PM
Screen Shot 2022-03-29 at 3 58 10 PM
Screen Shot 2022-03-29 at 3 58 16 PM
Screen Shot 2022-03-29 at 3 58 20 PM

# Links

[SPARK-313907](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-313907)
[Figma Article](https://www.figma.com/file/NaNrfXjygZtRgMfHAFHjsp/Components---Windows?node-id=31674%3A56400).

### CODE REVIEW

**Comment by chburket** [comment]
Nit: the screenshot for status color is showing "rounding = inactive". Is that an old shot?

**Comment by InteractiveTimmy** [comment]
> Nit: the screenshot for status color is showing "rounding = inactive". Is that an old shot?

This needed to be fixed, but i didn't notice it. Thanks! it's been updated/fixed!

**Comment by InteractiveTimmy** [comment]
I have updated the naming from `<MeetingCardContainer />` to `<CalendarCard />` to better match the design articles.

**Review by chburket — APPROVED** [review]
Looks good!

File: src/components/CalendarCard/CalendarCard.style.scss [hunk 1]
```diff
@@ -0,0 +1,71 @@
+.md-calendar-card-wrapper {
+  --local-normal-background-color: var(--theme-background-secondary-normal);
+  --local-hover-background-color: var(--theme-background-secondary-hover);
+  --local-active-background-color: var(--theme-background-secondary-active);
```
Comment (torgeadelin) [inline src/components/CalendarCard/CalendarCard.style.scss:4]: you can't use theme tokens here, you need to create new tokens in the tokens repo for this specific component, release, and use those instead.

**Review by torgeadelin — CHANGES_REQUESTED** [review]

**Comment by torgeadelin** [comment]
Also, still seeing both MeetingCardContainer and CalendarCard in storybook if i run it.

File: src/components/CalendarCard/CalendarCard.unit.test.tsx [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+import CalendarCard, { CALENDAR_CARD_CONSTANTS as CONSTANTS } from './';
+import { CALENDAR_CARD_STATUS_CONSTANTS as STATUS_CONSTANTS } from './CalendarCardStatus';
+
+describe('<CalendarCard />', () => {
+  describe('snapshot', () => {
+    it('should match snapshot', () => {
+      expect.assertions(1);
+
+      const { container } = render(<CalendarCard />);
```
[the remark this answers is not in the document]
Comment (InteractiveTimmy) [reply src/components/CalendarCard/CalendarCard.unit.test.tsx:13]: the `render()` method returns an object with a `container` key:value pair for testing. This needs to stay `container`.

File: src/components/CalendarCard/CalendarCard.unit.test.tsx [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+      const style = { color: 'pink' };
+
+      render(<CalendarCard style={style} data-testid={testid} />);
+
+      const component = await screen.findByTestId(testid);
+
+      expect(component.style.color).toBe(style.color);
+    });
+
+    it('should have child nodes that when children prop is provided', async () => {
```
[the remark this answers is not in the document]
Comment (InteractiveTimmy) [reply src/components/CalendarCard/CalendarCard.unit.test.tsx:158]: Fixed! good catch!

File: src/components/CalendarCard/CalendarCard.style.scss (hunk 1 above)
Comment (InteractiveTimmy) [reply src/components/CalendarCard/CalendarCard.style.scss:4]: Yep! Based on our design flow, I need this to be approved prior to generating the tokens, as if the design approval process here changes, we end up needing to make another PR on the tokens repo unnecessarily. Once approval is complete here, I'll make the changes to the token repo.

**Comment by InteractiveTimmy** [comment]
> Also, still seeing both MeetingCardContainer and CalendarCard in storybook if i run it.

Not sure why this is, I am not seeing this:

Screen Shot 2022-03-31 at 3 06 42 PM

**Review by torgeadelin — APPROVED** [review]
as discussed in 1:1 you will add the tokens and update this pr

**Comment by InteractiveTimmy** [comment]
This has been updated to accepted the new tokens in the tokens repo. as well as reflect the new design specifying `card` instead of `calendar card`.

**Pull request merged**